zebra-consensus/src/transaction.rs

```diff
@@ -87,6 +89,29 @@ pub enum Request {
     },
 }
 
+impl Request {
+    pub fn transaction(&self) -> &Arc<Transaction> {
+        match self {
+            Request::Block { transaction, .. } => transaction,
+            Request::Mempool { transaction, .. } => transaction,
+        }
+    }
+
+    pub fn known_utxos(&self) -> &Arc<HashMap<transparent::OutPoint, zs::Utxo>> {
```

Comment:
> I usually leave the decision to clone() the returned value or not to the caller, just in case they want to avoid the atomic increment and just use the reference directly, or in case they want a more ergonomic way of chaining calls without having to create a binding to keep the owned clone alive.

This is a good point.

I don't think we have a consistent style here.

Here are some of my thoughts so far:
- if we're calling a leaf function or method, we could just pass `&T`
- in general I prefer clear and correct code, sometimes getting rid of refs and derefs helps
- it would be nice to have a clippy lint to enforce whatever we decide

Reply:
That makes sense. Maybe it's hard to get a clear cut general rule for this :confused:  Maybe we can gather more input as we see more situations like this? Not sure if we should write it down somewhere :thinking: